staging/src/k8s.io/client-go/dynamic/dynamicinformer/informer.go

```diff
@@ -138,20 +153,79 @@ func NewFilteredDynamicInformer(client dynamic.Interface, gvr schema.GroupVersio
 			resyncPeriod,
 			indexers,
 		),
+		stopCh:  stopCh,
+		factory: factory,
 	}
 }
 
 type dynamicInformer struct {
 	informer cache.SharedIndexInformer
 	gvr      schema.GroupVersionResource
+	stopCh   chan struct{}
+	factory  *dynamicSharedInformerFactory
 }
 
 var _ informers.GenericInformer = &dynamicInformer{}
 
 func (d *dynamicInformer) Informer() cache.SharedIndexInformer {
-	return d.informer
+	return d
 }
 
 func (d *dynamicInformer) Lister() cache.GenericLister {
 	return dynamiclister.NewRuntimeObjectShim(dynamiclister.New(d.informer.GetIndexer(), d.gvr))
 }
+
+var _ cache.SharedIndexInformer = &dynamicInformer{}
+
+// It would be much cleaner if we could embed the struct cache.sharedIndexInformer in struct dynamicInformer
```

Comment:
Why not embedding cache.SharedIndexInformer?

Reply:
switched to embed the interface cache.SharedIndexInformer